segmentation/utils/spatial_analysis_utils.py

```diff
@@ -110,56 +114,65 @@ def compute_close_cell_num(dist_mat, dist_lim, num, analysis_type,
-                dist_mat_bin[np.ix_(
-                    np.asarray(mark1poslabels[j] - 1, dtype='int'),
-                    np.asarray(mark1poslabels[k] - 1, dtype='int')
-                )]
-            )
-            # symmetry :)
-            close_num[k, j] = close_num[j, k]
+            mark1poslabels = get_pos_cell_labels(analysis_type, thresh=thresh_vec.iloc[j],
+                                                 current_fov_channel_data=current_fov_channel_data,
+                                                 cell_labels=cell_labels,
+                                                 current_marker=current_fov_channel_data.columns[j])
+        # Length of the number of positive cell labels
+        marker1_poslabels_num.append(len(mark1poslabels))
+        for k in range(0, num):
+            # Repeats what was done above for the same marker and all other markers in the analysis
+            if analysis_type == "cluster":
+                mark2poslabels = get_pos_cell_labels(analysis_type, cluster_ids.iloc[k], current_fov_data)
+            else:
+                mark2poslabels = get_pos_cell_labels(analysis_type, thresh=thresh_vec.iloc[k],
+                                                     current_fov_channel_data=current_fov_channel_data,
+                                                     cell_labels=cell_labels,
+                                                     current_marker=current_fov_channel_data.columns[k])
+            marker2_poslabels_num.append(len(mark2poslabels))
+
+            # Subset the distance matrix to only include cells positive for both markers j and k
+            trunc_dist_mat = dist_mat[np.ix_(np.asarray(mark1poslabels - 1, dtype='int'),
+                                             np.asarray(mark2poslabels - 1, dtype='int'))]
 
-    return close_num, mark1_num
+            # Binarize the truncated distance matrix to only include cells within distance limit
+            trunc_dist_mat_bin = np.zeros(trunc_dist_mat.shape, dtype='int')
+            trunc_dist_mat_bin[trunc_dist_mat < dist_lim] = 1
+            close_num[j, k] = np.sum(np.sum(trunc_dist_mat_bin))
+    return close_num, marker1_poslabels_num, marker2_poslabels_num
 
 
-def compute_close_cell_num_random(marker_nums, dist_mat, dist_lim, bootstrap_num):
-    """Uses bootstrapping to permute cell labels randomly and records the number of close cells
-    (within the dit_lim) in that random setup.
+def compute_close_cell_num_random(marker1_poslabels_num, marker2_poslabels_num,
+                                  dist_mat, marker_num, dist_lim, bootstrap_num):
+    """Uses bootstrapping to permute cell labels randomly and records the number of close cells (within the dit_lim)
+    in that random setup.
 
     Args
-        marker_nums (np.array): list of cell counts of each marker type
-        dist_mat (np.array): cells x cells matrix with the euclidian
+        marker1_poslabels_num (list): number of cell labels for marker 1
```

Comment:
change to marker_nums (or change all in function references to marker1_poslabels_num to marker_nums)